src/openrct2/world/MoneyEffect.cpp

```diff
@@ -147,11 +147,12 @@ void MoneyEffect::Update()
 
 std::pair<rct_string_id, money32> MoneyEffect::GetStringId() const
 {
-    rct_string_id spentStringId = vertical ? STR_MONEY_EFFECT_SPEND_HIGHP : STR_MONEY_EFFECT_SPEND;
-    rct_string_id receiveStringId = vertical ? STR_MONEY_EFFECT_RECEIVE_HIGHP : STR_MONEY_EFFECT_RECEIVE;
+    rct_string_id spentStringId = CashFlow == TransactionType::CREDIT ? STR_MONEY_EFFECT_SPEND_HIGHP : STR_MONEY_EFFECT_SPEND;
+    rct_string_id receiveStringId = CashFlow == TransactionType::CREDIT ? STR_MONEY_EFFECT_RECEIVE_HIGHP
+                                                                        : STR_MONEY_EFFECT_RECEIVE;
```

Comment:
These lines show that `vertical` does not mean whether it credit/debit, IMO.

Reply:
I dug a bit further. The `HIGHP` strings contain currency strings with 2 decimals and are used for guest purchases. The only odd bit is why it increases Z for guest purchases.

Reply:
Guest purchases were added later so they work differently. I'd forgotten about that. Renaming the variable might not make sense now.